cid/helpers/randtime.py

```diff
@@ -0,0 +1,23 @@
+import re
+import random
+from datetime import datetime, timedelta
+
+
+def get_random_time_from_range(time_range):
+    """Return random time in format hh:mm within provided time range"""
+    if not time_range:
+        # return default random time between 02:00 and 04:59
+        return f"{random.randrange(2, 5):02d}:{random.randrange(0, 60):02d}", None
+
+    random_time = ""
+    try:
+        time_re = re.match(r'^(([01]\d|2[0-3]):([0-5]\d)|24:00)-(([01]\d|2[0-3]):([0-5]\d)|24:00)$', time_range)
```

Comment:
Could you please add comment explaining what is happening here and why we decided to do that

Reply:
We wanted to avoid that all scheduled refreshes run at the same time. Therefore, in resources.yaml having something like this 

```
schedules:
  default:
    id: cid
...
    schedule_frequency_time_range: '02:00-05:00'
```

would mean that scheduled refresh time will be randomly selected between 2am and 5am. Therefore, the regex from above expects that `schedule_frequency_time_range` is provided in hh:mm-hh:mm format. 


However, writing this now, I am thinking that maybe we can also provide possibility for fixed time (instead of time range), so to allow also this:

```
schedules:
  default:
    id: cid
...
    schedule_frequency_time_range: '02:45'
```

What do you think?

Reply:
This looks like a fix?
`if time_to < time_from: time_to +=  timedelta(days=1)`

Reply:
@iakov-aws  yes `if time_to < time_from: time_to +=  timedelta(days=1)` fixed it. Thanks!  Fix is pushed.

Reply:
Also i would try not to use regexp. You have 2 times, you can just split them and let str

```
timerange  = list (map(lambda x: x.strip(),  '23:09 - 02:22'.split('-')))
if len(timerange) == 1:
    return timerange[0]
elif len(timerange) == 2:
   ...
   # let strptime return a specific exception 
else:
   raise Exception('Input error ...')
```

Reply:
Let's discuss this. Regex helps when provided input is hh:mm-hh:mm-hh:mm or anything like that. Also with split, I would have to check if there are only two elements in the list. So, with regex I don't need this extra logic.

Reply:
All recommendations are implemented